server/issue.go

```diff
@@ -502,10 +502,31 @@ func (p *Plugin) httpGetJiraProjectMetadata(w http.ResponseWriter, r *http.Reque
 
 	instanceID := r.FormValue("instance_id")
 
-	plist, connection, err := p.ListJiraProjects(types.ID(instanceID), types.ID(mattermostUserID))
+	plist, connection, err := p.ListJiraProjects(types.ID(instanceID), types.ID(mattermostUserID), true)
 	if err != nil {
-		return respondErr(w, http.StatusInternalServerError,
-			errors.WithMessage(err, "failed to GetProjectMetadata"))
+		// Getting the issue Types separately only when the status code returned is 400
+		if strings.Contains(err.Error(), "400") {
```

Comment:
To limit the indentation below, we can return early here

```suggestion
		if !strings.Contains(err.Error(), "400") {
			return respondErr(w, http.StatusInternalServerError,
				errors.WithMessage(err, "failed to GetProjectMetadata"))
		}
```